src/lib/firestore-schema.ts

```diff
@@ -0,0 +1,152 @@
+ * As of this writing, the domain-related and supports-related aspects are already implemented, while
+ * the comment-related and network-related aspects are being planned.
+ * Where relevant, synchronization issues are noted. These are situations in which we will store
+ * information in Firestore that is replicated from elsewhere and which could potentially change,
+ * in which case we would need to have a mechanism for synchronizing that change. For instance,
+ * if a new teacher is added to a class, we would want to reflect that in our classes subcollection.
+ */
+// represents a Firestore subcollection
+type FSCollection<IDocument> = Record<string, IDocument>;
+// represents a Firestore timestamp
+type FSDate = Date;
+
+/*
+ * users
+ *
+ * Subcollection of domain with some fields copied from portal (e.g. name) with potential synchronization issues.
+ * At least initially, the list of networks a teacher is a part of will be maintained manually by editing in the
+ * Firestore console. It is anticipated that the numbers of teachers and networks will be small initially.
+ */
+interface UserDocument {
+  uid: string;                // portal user id
+  name: string;               // teacher name
+  type: string;               // "teacher" for now although maybe students eventually
+  networks: string[];         // ids of networks this teacher is part of
+}
+// collection key is portal uid
+type UsersCollection = FSCollection<UserDocument>;
+
+/*
+ * comments
+ *
+ * Subcollection of documents with a denormalized teacher name field with potential synchronization issues.
+ * Comments associated with a document are accessible to any user with access to the document.
+ */
+interface CommentDocument {
+  uid: string;                // user id of author of comment
+  name: string;               // [denormalized] name of user to avoid having to request user data separately
+  createdAt: FSDate;          // timestamp used for sorting
+  tileId?: string;            // empty for document contents
+  content: string;            // plain text for now; potentially html if we need rich text
+}
+// collection key is Firestore-assigned id
+type CommentsCollection = FSCollection<CommentDocument>;
+
+/*
+ * documents
+ *
+ * Subcollection of domain with several metadata fields copied from real-time database (e.g. title, properties)
+ * with potential synchronization issues.
+ * Documents and their comments are accessible to the owning teacher in addition to any network access.
+ * Documents owned by one teacher are accessible to all teachers that share a network with the owner.
+ * Networked access will be mediated by a Firebase function which can check whether the requesting user and
+ * the owning user share a network.
+ */
+interface DocumentDocument {
+  uid: string;
+  type: string;
+  key: string;
+  createdAt: FSDate;
+  title?: string;
+  properties: Record<string, string>;
+  comments: CommentsCollection;
```

Comment:
My thinking is that the primary/only comments query will be to get all of the comments for a particular document. Plus, I want access to the comments to be controlled by access to the document, i.e. if you have access to the document through the network then you have access to all of the comments attached to the document. Otherwise, you run the risk of getting into situations in which some comments aren't viewable by those who can view the document and vice-versa, which would just lead to unintelligible comments threads. Furthermore, I think that if a teacher makes a comment on a document while in a network, and then that teacher is removed from that network, the comment should still be viewable by those in the network. In other words, the comment becomes "network-visible" when it is made by a teacher in the network and changing the status of the teacher doesn't automatically change the status if all of the teacher's comments.